### PR TITLE
Feat: add textbox to enter confirmation number

### PR DESCRIPTION
**Pull request by akshatchhabra**
These changes add a textbox where a user can enter a confirmation number to confirm an alternate email.

### CODE REVIEW

**Comment by melisabok** [comment]
Akshat, some test are failing here. I think it is related to your changes.

**Comment by akshatchhabra** [comment]
token verification box

invalid token

token verified

**Comment by melisabok** [comment]
can the textbox and the verify button be in the same line than then email address? we can remove the confirm and remove button to have more space.

**Comment by akshatchhabra** [comment]
> can the textbox and the verify button be in the same line than then email address? we can remove the confirm and remove button to have more space.

I wonder if the user might want to request the code again or choose not to add the email. They might also want to use the confirm and remove button.

**Comment by carlosmondra** [comment]
> > can the textbox and the verify button be in the same line than then email address? we can remove the confirm and remove button to have more space.
> 
> I wonder if the user might want to request the code again or choose not to add the email. They might want to use the confirm and remove button.

Maybe we should remove the confirm button only. If they want a new code, they can refresh the page to request a new code.

**Comment by melisabok** [comment]
> Maybe we should remove the confirm button only. If they want a new code, they can refresh the page to request a new code.

I agree.

**Comment by akshatchhabra** [comment]
I rearranged it and hid the confirm button.
rearranged

**Comment by akshatchhabra** [comment]
I found a problem, if I add multiple emails and click on confirm for only one it shows the verification token input for both:
Screenshot 2024-06-05 at 12 11 50 PM
Woking on a way to fix this

**Comment by carlosmondra** [comment]
I think you already found this issue. When there is more than one field to input an email the same number of verify text boxes appear even if the text box is empty.

ScreenRecording2024-06-05at3 28 47PM-ezgif com-video-to-gif-converter

**Comment by carlosmondra** [comment]
Another issue that I found is that after clicking on Verify, we should remove all the buttons.

**Comment by melisabok** [comment]
I would add the text box and verify button first and delete button last.

File: components/profile/EmailsSection.js [hunk 1]
```diff
@@ -63,6 +81,9 @@ const EmailsSection = ({
   institutionDomains,
   isNewProfile,
 }) => {
+  const [isVerifyTextboxVisible, setIsVerifyTextboxVisible] = useState(false)
+  const [verificationToken, setVerificationToken] = useState('')
```
[the remark this answers is not in the document]
Comment (akshatchhabra) [reply components/profile/EmailsSection.js:75]: This helped, thanks for the suggestion.
I've pushed the changes

File: components/profile/EmailsSection.js [hunk 2]
```diff
@@ -165,25 +237,48 @@ const EmailsSection = ({
...
+            { emailObj.visible &&
+              <div className='col-md-1 emails__value'>
+                  <EmailsButton
+                    type="verify"
+                    emailObj={emailObj}
+                    handleVerify={() => handleVerifyEmail(emailObj.key)}
+                  />
+              </div>
+            }
+            { !(emailObj.verified && !emailObj.confirmed) && !emailObj.visible  &&
```
[the remark this answers is not in the document]
Comment (akshatchhabra) [reply components/profile/EmailsSection.js]: Fixed

File: components/profile/EmailsSection.js [hunk 3]
```diff
@@ -124,16 +164,47 @@ const EmailsSection = ({
...
       const linkData = { alternate: newEmail, username: profileId }
       try {
-        await api.post('/user/confirm', linkData, { accessToken })
-        return promptMessage(`A confirmation email has been sent to ${newEmail}`)
+        if (isNewProfile) {
+          await api.post(`/user/confirm/${router.query.token}`, linkData, { accessToken })
+        } else {
+          await api.post('/user/confirm', linkData, { accessToken })
+        }
+        setEmails({ setVisible: true, data: { key, visibleValue: true} })
```
[the remark this answers is not in the document]
Comment (akshatchhabra) [reply components/profile/EmailsSection.js]: changed the variable name

**Review by melisabok — APPROVED** [review]

**Review by carlosmondra — APPROVED** [review]

**Pull request merged**